src/main_functions.c

```diff
@@ -0,0 +1,478 @@
+#include <stdio.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <stdlib.h>
+#include <string.h>
+#include <assert.h>
+#include "shellstrings.h"
+#include "parser.h"
+#include "main_functions.h"
+#include "dictionary.h"
+#include "word.h"
+#include <stdlib.h>	
+
+
+
+/* 
+ *	Order of functions:
+ *		I. Saving files
+ *		II. Functions for editing strings (lines)
+ *		III. Interactive Mode
+ *		IV. Batch Mode
+ *		V. Main Page
+ */
+
+
+/*
+ *	I. Saving Files
+ */
+void save_corrections(char* filename, char** lines)
+{
+	FILE* f = fopen(filename,"w");
+    assert(f != NULL);
+	int i = 0;
+	while (lines[i] != NULL) {
+		fprintf(f, "%s", lines[i]); // write lines into file
+		i++;
+	}
+	fclose(f);
+}
+
+void save_page(char* filename, char** lines, int* quit)
+{
+	int i = 1;
+    char line[256];
+    char* args = NULL;
+    int verify = 0;
+
+	while (i) {
+        shell_save();
+		shell_prompt();
+		i = 0;
+
+        verify = scanf("%s", line);
+        assert (!(verify < 0)); // ensure valid input
+        assert (strlen(line) < 256); // ensure that string length is within limit
+
+        if (strlen(line) > 2) { // ensure that only one character is entered, otherwise cannot proceed
+            shell_error("Please type in one of the indicated commands!");
+            i = 1;
+        } else if (!strcmp(line, "s")) {
+			save_corrections(filename, lines);  // save to the same file destination, overwriting existing file
+			*quit = 1;
+		} else if (!strcmp(line, "c")) {
+            
+            while ((args == NULL) ^ i) {    // either user returns to main page, or inputs a vaid new destination
+                printf("\n\nEnter a viable file name (*.txt), or enter 'r' to return to the save page.\n\n");
+                shell_prompt();
+                verify = scanf("%s", line);
+                assert (!(verify < 0));
+
+                args = strstr(line, ".txt");
+                if (!strcmp(line, "r")) i = 1;
+            }
+
+            *quit = 1;
+            if (i == 0) save_corrections(line, lines);  // save to different file destination
+
+		} else if (!strcmp(line, "r")) {
+			*quit = 0;
+		} else if (!strcmp(line, "q")) {
+			*quit = 1;
+		} else {
+			shell_error("Please type in one of the indicated commands!");   // wrong input
+			i = 1;
+		}
+
+	}
+}
+
+/*
+ *	II. Functions for editing strings
+ */
+
+void underline_misspelled(char *word, char* underline)
+{
+	int j = strlen(word);
+    int i;
+	for(i = 0; i < j; i++) {
+    	strcat(underline, "^");
+	}
+		strcat(underline, " ");
```

Comment:
@sarikam1

Reply:
One note i'd add on this is that to implement a non-strcat version of this function, it would be necessary to add an additional parameter as a result of the fact that this function is called in parse_string to accumulate alternating "^"s and " "s (the latter being added with underline_correct). Hence, strcat may be more convenient.

Reply:
Ok, that's fine then